src/internal/common/prefix_matcher.go

```diff
@@ -0,0 +1,75 @@
+package common
```

Comment:
I feel nitpicky about "common" these days.  Thoughts on either of these?
1. adding this to Filters
2. putting it in a subpackage: common/prefix_matcher